consensus/src/types.rs

```diff
@@ -1,73 +1,400 @@
+/// It is used for epoch arithmetic operations and defining epoch bounds for data retention.
+pub type EpochDelta = Delta<Epoch>;
+
+/// Type alias for view offsets and durations.
+///
+/// `ViewDelta` represents a distance between views or a duration measured in views.
+/// It is commonly used for timeouts, activity tracking windows, and view arithmetic.
+pub type ViewDelta = Delta<View>;
+
+/// A unique identifier combining epoch and view for a consensus round.
+///
+/// Round provides a total ordering across epoch boundaries, where rounds are
+/// ordered first by epoch, then by view within that epoch.
 #[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct Round(Epoch, View);
+pub struct Round {
+    epoch: Epoch,
+    view: View,
+}
 
 impl Round {
-    pub fn new(epoch: Epoch, view: View) -> Self {
-        Self(epoch, view)
+    /// Creates a new round from an epoch and view.
+    pub const fn new(epoch: Epoch, view: View) -> Self {
+        Self { epoch, view }
     }
 
-    pub fn epoch(&self) -> Epoch {
-        self.0
+    /// Returns round zero, i.e. epoch zero and view zero.
+    pub const fn zero() -> Round {
+        Round::new(Epoch::zero(), View::zero())
     }
 
-    pub fn view(&self) -> View {
-        self.1
+    /// Returns the epoch of this round.
+    pub fn epoch(self) -> Epoch {
+        self.epoch
+    }
+
+    /// Returns the view of this round.
+    pub fn view(self) -> View {
+        self.view
     }
 }
 
 impl From<(Epoch, View)> for Round {
     fn from((epoch, view): (Epoch, View)) -> Self {
-        Self(epoch, view)
+        Self { epoch, view }
     }
 }
 
 impl From<Round> for (Epoch, View) {
     fn from(round: Round) -> Self {
-        (round.epoch(), round.view())
+        (round.epoch, round.view)
     }
 }
 
 impl Read for Round {
     type Cfg = ();
 
     fn read_cfg(buf: &mut impl Buf, _cfg: &Self::Cfg) -> Result<Self, Error> {
-        Ok(Self(Epoch::read(buf)?, View::read(buf)?))
+        Ok(Self {
+            epoch: Epoch::read(buf)?,
+            view: View::read(buf)?,
+        })
     }
 }
 
 impl Write for Round {
     fn write(&self, buf: &mut impl BufMut) {
-        self.epoch().write(buf);
-        self.view().write(buf);
+        self.epoch.write(buf);
+        self.view.write(buf);
     }
 }
 
 impl EncodeSize for Round {
     fn encode_size(&self) -> usize {
-        self.epoch().encode_size() + self.view().encode_size()
+        self.epoch.encode_size() + self.view.encode_size()
     }
 }
 
 impl Display for Round {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "Round({}, {})", self.0, self.1)
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "({}, {})", self.epoch, self.view)
+    }
+}
+
+/// An iterator over a range of views.
+///
+/// Created by [`View::range`]. Iterates from start (inclusive) to end (exclusive).
+pub struct ViewRange {
+    current: View,
```

Comment:
prefer `start` rather than `current` since it may be a "backwards-facing" range? Like the "current" might actually be at the end? I guess it doesn't matter much since it's private fields but would just be more consistent

Is there a reason this is better than just using [Range](https://doc.rust-lang.org/std/ops/struct.Range.html) over `View`?